js/ui/map.js

```diff
@@ -238,6 +240,8 @@ class Map extends Camera {
         } else {
             positionContainer.appendChild(controlElement);
         }
+        this._controls[control.constructor.name] = control;
```

Comment:
`Function#name` has many caveats (see the [MDN compatibility table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name)). It will be safer to use a more explicit naming system or eliminate the need for a naming system.